### DISCUSSION TITLE
HAProxy with ScadaLTS

### DISCUSSION THREAD

**Discussion by hjbleier** [SCADA-LTS/Scada-LTS#2817] [Q&A]
Hi Everyone,

I got a small issue, it's not a direct ScadaLTS issue but I wanted to see if there is anyone that has experienced this same problem. I use HAProxy on my firewall (pfSense) to handle all of my SSL certs, I have a very large amount of virtual machines in a high availability cluster and I find it must easier to manage all of the SSL certs in one place, so when there is a issue it is very easy to correct it quickly. Anyways so I am running my ScadaLTS instance thru HAProxy and I keep getting this popup that says "Connect error:Whoops! Lost connection to https://scadalts.xxxxxxxx.com/scadalts/ws-scada/alarmLevel". I have looked thru the HAProxy config and I can not figure it out. If I connect locally to the virtual server running ScadaLTS with no SSL it works as normal and get no popups. The popup seems to be randomly popping up, sometimes varying from every few seconds to up-to every two minute or so. The big problem is you can't leave a page open to monitor something because when the popup does occur the live values stop until you click the okay button.
Is there anyone that has their ScadaLTS running thru HAProxy? If so and you got it working good, can you be so kind as to post your HAProxy config (blocking out your private info, ip, domain, etc.) please. This way I will have a starting point to hopefully get mine working correctly without this popup. Thank You!

**Comment by rafaelpsousa** [comment]
I had this message error with Apache Reverse Proxy, to resolve i put the ssl certificate in apache config and tomcat

```
Apache Virtual Host:

<VirtualHost *:443>
    ServerName [scada.xpto.br](http://scada.xpto.br/)

   # Proxy HTTP normal
   ProxyPass "/" "https://scada.xpto.br:8443/"
   ProxyPassReverse "/" "https://scada.xpto.br:8443/"

   # Proxy  WebSocket
   RewriteEngine on
   RewriteCond %{HTTP:Upgrade} =websocket [NC]
   RewriteCond %{HTTP:Connection} upgrade [NC]
   RewriteRule ^/?(.*) "wss://[scada.xpto.br:8443/$1](http://scada.xpto.br:8443/$1)" [P,L]

    SSLEngine on
    SSLCertificateFile "/usr/local/etc/apache24/certs/xpto-br.crt"
    SSLCertificateKeyFile "/usr/local/etc/apache24/certs/private.key"
    SSLCertificateChainFile "/usr/local/etc/apache24/certs/gs_root_and_intermediate.crt"

    SSLProxyEngine on

</VirtualHost>


In the /etc/hosts file, I added an entry for the internal Scada-LTS server:

10.0.0.123 [scada.xpto.br](http://scada.xpto.br/)


Tomcat server.xml

<Connector port="8443" protocol="org.apache.coyote.http11.Http11NioProtocol"
  maxThreads="150" SSLEnabled="true">
<SSLHostConfig>
<Certificate certificateKeystoreFile="conf/keystore.jks" certificateKeystorePassword="password"
type="RSA" />
</SSLHostConfig>
</Connector>
